### PR TITLE
fix: det job get returns inaccurate info

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Fixes two bugs:
- bug where the reported priority was always the default
- bug where master panics when calling `det job list`

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/resourcemanagers/kubernetes_resource_manager.go [hunk 1]
```diff
@@ -205,9 +205,15 @@ func (k *kubernetesResourceManager) receiveRequestMsg(ctx *actor.Context) error
 	case sproto.SetGroupMaxSlots:
 		k.getOrCreateGroup(ctx, msg.Handler).maxSlots = msg.MaxSlots
 
-	case sproto.SetGroupWeight, sproto.SetGroupPriority:
+	case sproto.SetGroupWeight:
 		// SetGroupWeight and SetGroupPriority are not supported by the Kubernetes RP.
 
+	case sproto.SetGroupPriority:
```
Comment (hamidzr) [inline master/internal/resourcemanagers/kubernetes_resource_manager.go:211]: isn't this a write side feature?

File: master/internal/resourcemanagers/priority.go [hunk 2]
```diff
@@ -383,3 +383,39 @@ func taskFilter(label string, zeroSlots bool) func(*sproto.AllocateRequest) bool
...
+	var reqs []*sproto.AllocateRequest
+
+	for it := taskList.iterator(); it.next(); {
+		reqs = append(reqs, it.value())
+	}
+
+	sort.Slice(reqs, func(i, j int) bool {
+		p1 := *groups[reqs[i].Group].priority
+		p2 := *groups[reqs[j].Group].priority
+		if k8s { // in k8s, higher priority == more prioritized
```
Comment (hamidzr) [inline master/internal/resourcemanagers/priority.go:401]: nit I think you can flip p1 and p2 value like we do in api_job for a bit shorter syntax

File: master/internal/resourcemanagers/priority.go [hunk 3]
```diff
@@ -383,3 +383,39 @@ func taskFilter(label string, zeroSlots bool) func(*sproto.AllocateRequest) bool
 		return request.Label == label && (request.SlotsNeeded == 0) == zeroSlots
 	}
 }
+
+func sortTasks(
```
Comment (hamidzr) [inline master/internal/resourcemanagers/priority.go:387]: nit: will this be used in the agent-based priority scheduler?

Comment (eecsliu) [reply master/internal/resourcemanagers/priority.go:387]: not at the moment - the agent one sets the job status, whereas in k8s we must observe the pod status

File: master/internal/resourcemanagers/kubernetes_resource_manager.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/resourcemanagers/kubernetes_resource_manager.go:211]: this is necessary for determined to adjust the priority from the default to what the user sets in the config.

**Review by hamidzr — APPROVED** [review]

**Pull request merged**